pkg/helpers/helpers.go

```diff
@@ -33,3 +34,24 @@ func IsObjectTaggedToDelete(obj client.Object) bool {
 	annotation, ok := annotations[DeleteTagAnnotation]
 	return ok && annotation == "true"
 }
+
+func MergeLabelMap(existing *map[string]string, desired map[string]string) bool {
```

Comment:
This method could be used for labels, but also for Annotations. In general, for any `map[string]string`.  Consider changing the name to `MergeMapStringString`